app/controllers/comments_controller.rb

```diff
@@ -8,10 +10,26 @@ def create
     redirect_to share_path(@share)
   end
 
+  def destroy
+   return render_not_found if @comment.blank?
+   return render_not_found(:forbidden) if @comment.user != current_user
+   @comment.destroy
+   flash[:notice] = "Your comment was deleted."
+   redirect_to :back
+  end
+
   private
 
-  def comment_params
-    params.require(:comment).permit(:body)
-  end
+    def comment_params
+      params.require(:comment).permit(:body)
+    end
+
+    def set_comment
+      @comment = Comment.find(params[:id])
+    end
+
+    def require_creator
+      access_denied unless user_signed_in? and (current_user == @comment.user)
```

Comment:
I see. I thought their was a subtle difference, didn't realize it was that serious.